cpp/src/arrow/compute/kernels/aggregate_quantile.cc

```diff
@@ -524,23 +530,13 @@ void AddQuantileKernels(VectorFunction* func) {
     base.signature = KernelSignature::Make({InputType(ty)}, OutputType(ResolveOutput));
     // output type is determined at runtime, set template argument to nulltype
     base.exec = GenerateNumeric<QuantileExecutor, NullType>(*ty);
-    base.exec_chunked =
-        GenerateNumeric<QuantileExecutorChunked, NullType, VectorKernel::ChunkedExec>(
-            *ty);
-    DCHECK_OK(func->AddKernel(base));
-  }
-  {
-    base.signature =
-        KernelSignature::Make({InputType(Type::DECIMAL128)}, OutputType(ResolveOutput));
-    base.exec = QuantileExecutor<NullType, Decimal128Type>::Exec;
-    base.exec_chunked = QuantileExecutorChunked<NullType, Decimal128Type>::Exec;
+    base.exec_chunked = GenerateNumeric<QuantileExecutorChunked, NullType>(*ty);
     DCHECK_OK(func->AddKernel(base));
   }
-  {
-    base.signature =
-        KernelSignature::Make({InputType(Type::DECIMAL256)}, OutputType(ResolveOutput));
-    base.exec = QuantileExecutor<NullType, Decimal256Type>::Exec;
-    base.exec_chunked = QuantileExecutorChunked<NullType, Decimal256Type>::Exec;
+  for (auto type_id : DecimalTypeIds()) {
```

Comment:
For the record, this is actually extending the supported decimal types by `Decimal32` and `Decimal64` that is independent of function `winsorize`?

Reply:
Not entirely independent, since it's necessary for `winsorize` itself to work on those types
(and it's mentioned in the PR description :)).

Reply:
Ah, I see. Thank you.